src/libs/antares/study/parallel-years.cpp

```diff
@@ -0,0 +1,426 @@
+        numberOfMCYearThreads[ncAvg] = 5;
+        numberOfMCYearThreads[ncHigh] = 8;
+        numberOfMCYearThreads[ncMax] = 9;
+        break;
+    case 11:
+        numberOfMCYearThreads[ncLow] = 3;
+        numberOfMCYearThreads[ncAvg] = 6;
+        numberOfMCYearThreads[ncHigh] = 8;
+        numberOfMCYearThreads[ncMax] = 10;
+        break;
+    case 12:
+        numberOfMCYearThreads[ncLow] = 3;
+        numberOfMCYearThreads[ncAvg] = 6;
+        numberOfMCYearThreads[ncHigh] = 9;
+        numberOfMCYearThreads[ncMax] = 11;
+        break;
+    default:
+        numberOfMCYearThreads[ncLow] = (uint)std::ceil(nbLogicalCores / 4.);
+        numberOfMCYearThreads[ncAvg] = (uint)std::ceil(nbLogicalCores / 2.);
+        numberOfMCYearThreads[ncHigh] = (uint)std::ceil(3 * nbLogicalCores / 4.);
+        numberOfMCYearThreads[ncMax] = nbLogicalCores - 1;
+        break;
+    }
+
+    /*
+            Getting the number of parallel years based on the number
+            of cores level.
+            This number is limited by the smallest refresh span (if at least
+            one type of time series is generated)
+    */
+
+    try
+    {
+        forcedNbOfParallelYears = numberOfMCYearThreads.at(p.nbCores.ncMode);
+    }
+    catch(const std::out_of_range& e)
+    {
+        logs.fatal() << "Simulation cores level not correct : " << (int)p.nbCores.ncMode;
+    }
+    
+}
+
+
+void SetsOfParallelYearCalculator::limitNbOfParallelYearsbyMinRefreshSpan()
+{
+    uint TSlimit = UINT_MAX;
+    if ((p.timeSeriesToGenerate & timeSeriesLoad) && (p.timeSeriesToRefresh & timeSeriesLoad))
+        TSlimit = p.refreshIntervalLoad;
+    if ((p.timeSeriesToGenerate & timeSeriesSolar) && (p.timeSeriesToRefresh & timeSeriesSolar))
+        TSlimit = (p.refreshIntervalSolar < TSlimit) ? p.refreshIntervalSolar : TSlimit;
+    if ((p.timeSeriesToGenerate & timeSeriesHydro) && (p.timeSeriesToRefresh & timeSeriesHydro))
+        TSlimit = (p.refreshIntervalHydro < TSlimit) ? p.refreshIntervalHydro : TSlimit;
+    if ((p.timeSeriesToGenerate & timeSeriesWind) && (p.timeSeriesToRefresh & timeSeriesWind))
+        TSlimit = (p.refreshIntervalWind < TSlimit) ? p.refreshIntervalWind : TSlimit;
+    if ((p.timeSeriesToGenerate & timeSeriesThermal) && (p.timeSeriesToRefresh & timeSeriesThermal))
+        TSlimit = (p.refreshIntervalThermal < TSlimit) ? p.refreshIntervalThermal : TSlimit;
+
+    if (TSlimit < forcedNbOfParallelYears)
+        forcedNbOfParallelYears = TSlimit;
+
+    // Limiting the number of parallel years by the total number of years
+    if (p.nbYears < forcedNbOfParallelYears)
+        forcedNbOfParallelYears = p.nbYears;
+}
+
+bool SetsOfParallelYearCalculator::isRefreshNeededForCurrentYear(uint y)
+{
+    bool refreshing = false;
+    refreshing = (p.timeSeriesToGenerate & timeSeriesLoad)
+                    && (p.timeSeriesToRefresh & timeSeriesLoad)
+                    && (!y || ((y % p.refreshIntervalLoad) == 0));
```

Comment:
```suggestion
                    && ((y % p.refreshIntervalLoad) == 0);
```
since `!y` implies `(y % p.refreshIntervalLoad) == 0`, this is equivalent.